chispa/dataframe_comparer.py

```diff
@@ -5,14 +5,22 @@
 import chispa.six as six
 from functools import reduce
 
+
 class DataFramesNotEqualError(Exception):
    """The DataFrames are not equal"""
    pass
 
 
-def assert_df_equality(df1, df2, ignore_nullable = False, transforms = [], allow_nan_equality = False):
-    df1 = reduce(lambda acc, fn: fn(acc), transforms, df1)
-    df2 = reduce(lambda acc, fn: fn(acc), transforms, df2)
+def assert_df_equality(df1, df2, ignore_nullable=False, transforms=None, allow_nan_equality=False, ignore_column_order=False, ignore_row_order=False):
```

Comment:
`ignore_row_order` is an interesting option. AFAIK DataFrames are orderless unless they have an explicit `.orderBy()` tacked on to their plan. This is analogous to [tables in SQL](https://dba.stackexchange.com/a/6053/2660), btw.

So wouldn't you want the default to be `ignore_row_order=True`?

Reply:
When `ignore_row_order` is set to `True` then both DataFrames are sorted before they're compared.  This is slower and potentially less intuitive (cause the DataFrames in your test suite have a different order than the outputted error message).  I think this is an important feature, especially for operations that can return results in random orders, but shouldn't be the default.  Let's chat about this more.

Reply:
Maybe I'm the outlier, but I consider the more intuitive check -- especially for testing purposes -- to ignore order. If some function produces a DataFrame that I want to check, I care about the contents. And by default, Spark offers no guarantees on row order unless your plan has an explicit `.orderBy()`. So relying on the stability of row order in the absence of an explicit order by clause is a recipe for surprises, much like it is in SQL.

In fact, I don't think `.collect()` even provides any guarantees that the row order of the resulting array will match the row order of the original DataFrame---again, unless the DataFrame has an explicit ordering specified. It's theoretically possible, for example, that you could call `spark.range(3).collect()` twice and get different row orders each time. So if you're relying on `.collect()` to preserve order without explicit ordering on the original DataFrames, then I would say that's technically incorrect.

By the way, in your own usages of this library (or the Scala equivalent), how often do you compare DataFrames where you care about the row order? I'm curious to see a few examples of that.

Reply:
Created a separate issue to explore this suggestion.